web-app/HeimScripts/heatmap/run.r

```diff
@@ -7,6 +7,8 @@ main <- function(max_rows=50){
   if(ncol(df) > 3){
     variances <- apply(df[,3:ncol(df)],1,var) # Calculating variance per probe
     df["SIGNIFICANCE"] <- variances
+    df["MEAN"] <- apply(df[,3:ncol(df)],1,mean, na.rm = T)
```

Comment:
Bug alert: you add another column when you calculate the variance. This means that for each you are now calculating the mean over the combination of the expression values and variance, and you are calculating the standard deviation over the combination of expression values, variance and mean.
Run the following:
df  <- data.frame(Row.Label= letters[1:5] , Bio.marker=letters[2:6], z=11:15, u=26:30)
if(ncol(df) > 3){
     variances <- apply(df[,3:ncol(df)],1,var) # Calculating variance per probe
     df["SIGNIFICANCE"] <- variances
     df["MEAN"] <- apply(df[,3:ncol(df)],1,mean, na.rm = T)
     df["SD"] <- apply(df[,3:ncol(df)],1,sd, na.rm = T)
     df <- df[with(df, order(-SIGNIFICANCE)), ]
}
df

And compare with:
df  <- data.frame(Row.Label= letters[1:5] , Bio.marker=letters[2:6], z=11:15, u=26:30)

apply(df[, 3:4], FUN=var, MARGIN=1)
apply(df[, 3:4], FUN=mean, MARGIN=1)
apply(df[, 3:4], FUN=sd, MARGIN=1)